crates/ide-completion/src/completions/attribute.rs

```diff
@@ -82,16 +82,15 @@ pub(crate) fn complete_attribute(acc: &mut Completions, ctx: &CompletionContext)
     };
 
     match qualified {
-        Qualified::With { resolution, is_super_chain, .. } => {
+        Qualified::With {
+            resolution: Some(hir::PathResolution::Def(hir::ModuleDef::Module(module))),
+            is_super_chain,
+            ..
+        } => {
             if *is_super_chain {
                 acc.add_keyword(ctx, "super::");
             }
 
-            let module = match resolution {
-                Some(hir::PathResolution::Def(hir::ModuleDef::Module(it))) => it,
-                _ => return,
-            };
```

Comment:
This would mean `super::` doesn't get added as completion in certain cases, right?

Reply:
Oh, good catch that is indeed the case

Reply:
Actually this should be fine, since the qualifier can only resolve into a module if it consists only of `super` segments (and if there are no segments to resolve we won't hit this branch anyways)